src/libexpr/eval.cc

```diff
@@ -1486,11 +1517,32 @@ class CallDepth {
 };
 };
 
+namespace {
+class SampleStack {
+    EvalState & state;
+public:
+    SampleStack(EvalState & state, const PosIdx pos) : state(state) {
+        state.stack.push_back(pos);
+    }
+    ~SampleStack() {
+        auto now = std::chrono::high_resolution_clock::now();
+        if (now - state.lastStackSample > SAMPLE_INTERVAL) {
```

Comment:
This technically slows down all calls, but probably not in a way that biases the result in a significant way.
Nonetheless, we'll need to know the performance impact of this.

Possible mitigations:
- add a conditional `if (evalSettings.profiling)`; a predictable branch is cheaper than getting the time
- conditional compilation: worse UX but guaranteed to work

So, specifically what we need is benchmarks of
- the control: `git merge-base HEAD upstream/master`
- the current unconditional implementation
- an implementation with a conditional based on a setting, with the setting disabled
- not critical, but good to know: a measurement with the setting enabled